modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -31,29 +32,42 @@ func (a TransferAuthorization) Accept(ctx sdk.Context, msg sdk.Msg) (authz.Accep
 		return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidType, "type mismatch")
 	}
 
+	var (
+		limitLeft  = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(math.MaxInt64)))
+		isNegative bool
+	)
+
 	for index, allocation := range a.Allocations {
 		if !(allocation.SourceChannel == msgTransfer.SourceChannel && allocation.SourcePort == msgTransfer.SourcePort) {
 			continue
 		}
 
-		limitLeft, isNegative := allocation.SpendLimit.SafeSub(msgTransfer.Token)
-		if isNegative {
-			return authz.AcceptResponse{}, errorsmod.Wrapf(ibcerrors.ErrInsufficientFunds, "requested amount is more than spend limit")
-		}
+		for _, coin := range allocation.SpendLimit {
+			// skip spending limit if the amount is set to max int64
+			if coin.Amount.Int64() == math.MaxInt64 {
+				continue
+			}
```

Comment:
Now that we `continue` here we should move the `isAllowedAddress()` check to before this statement. Perhaps even outside of the scope of this coin iteration